src/citrine/informatics/predictors/mean_property_predictor.py

```diff
@@ -36,36 +44,42 @@ class MeanPropertyPredictor(
         If ``True`` and a default is specified in ``default_properties``, then the specified
         default is used in place of missing values.
     label: Optional[str]
-        Only ingredients with this label will be counted in calculating the generalized mean.
+        Only ingredients with this label are counted when calculating the component-weighted mean.
         If ``None`` (default) all ingredients will be counted.
+    default_properties: Optional[Mapping[str, Union[str, float]]]
+        Default values to use for imputed properties.
+        Defaults are specified as a map from descriptor key to its default value.
+        If not specified and ``impute_properties == True`` the average over the entire dataset
+        will be used to fill in missing values. Any specified defaults will be used in place of
+        the average over the dataset. ``impute_properties`` must be ``True`` if
+        ``default_properties`` are provided.
     training_data: Optional[List[DataSource]]
         Sources of training data. Each can be either a CSV or an GEM Table. Candidates from
         multiple data sources will be combined into a flattened list and de-duplicated by uid and
         identifiers. De-duplication is performed if a uid or identifier is shared between two or
         more rows. The content of a de-duplicated row will contain the union of data across all
         rows that share the same uid or at least 1 identifier. Training data is unnecessary if the
         predictor is part of a graph that includes all training data required by this predictor.
-    default_properties: Optional[Mapping[str, float]]
-        Default values to use for imputed properties.
-        Defaults are specified as a map from descriptor key to its default value.
-        If not specified and ``impute_properties == True`` the average over the entire dataset
-        will be used to fill in missing values. Any specified defaults will be used in place of
-        the average over the dataset. ``impute_properties`` must be ``True`` if
-        ``default_properties`` are provided.
 
     """
 
     _resource_type = ResourceTypeEnum.MODULE
 
     input_descriptor = _properties.Object(FormulationDescriptor, 'config.input')
-    properties = _properties.List(_properties.Object(RealDescriptor), 'config.properties')
-    p = _properties.Integer('config.p')
-    training_data = _properties.List(_properties.Object(DataSource),
-                                     'config.training_data', default=[])
+    properties = _properties.List(_properties.Object(Descriptor), 'config.properties')
```

Comment:
I tried to make this 
```
properties = _properties.List(
_properties.Union([_properties.Object(RealDescriptor), _properties.Object(CategoricalDescriptor)]),
'config.properties'
)
```
but it was failing deserialization of platform objects. 

Do you know how to make that work instead of just the polymorphic object? I think we'd rather be explicit.

Reply:
https://github.com/CitrineInformatics/citrine-python/blob/177795d6408d9667d053a25e2f26a3b4a079bbcd/src/citrine/informatics/predictors/graph_predictor.py#L37-L38
https://github.com/CitrineInformatics/citrine-python/blob/177795d6408d9667d053a25e2f26a3b4a079bbcd/src/citrine/informatics/design_spaces/product_design_space.py#L35-L37

Looks consistent with other usages.  Might the default be necessary for something about the endpoint?

Reply:
I think I solved it in this next commit. There was an issue in the serialization pckage where a `RuntimeError` was being thrown instead of a `ValueError`, which caused the `Union` type to fail during deserialization.

Reply:
I would expect what you tried to work. Were you able to track down the root cause of the deserialization failure?

Reply:
The error is occurring because `properties.Union` attempts to deserialize w/ the element types in order in a try/catch block. When those element types are `properties.Object`, we end up calling `Property.deserialize_from_dict` which raises a `RuntimeError` on line 93.

This gets a bit iffy. That _should_ probably be a `RuntimeError`, so catching both `ValueError` and `RuntimeError` inside `properties.Union` may be the best approach.

Reply:
ehh maybe line 93 could logically also be a `ValueError`. Switching that to `ValueError` breaks a few other tests, though, which I'll have to trouble shoot.